bson/src/main/org/bson/Document.java

```diff
@@ -259,6 +261,63 @@ public Date getDate(final Object key) {
         return (Date) get(key);
     }
 
+    /**
+     * Gets the list value of the given key, casting the list elements to the given {@code Class<T>}.  This is useful to avoid having
+     * casts in client code, though the effect is the same.
+     *
+     * @param key   the key
+     * @param clazz the non-null class to cast the list value to
+     * @param <T>   the type of the class
+     * @return the list value of the given key, or null if the instance does not contain this key.
+     * @throws ClassCastException if the elements in the list value of the given key is not of type T or the value is not a list
+     */
+    public <T> List<T> getList(final Object key, final Class<T> clazz) {
+        notNull("clazz", clazz);
+        return constructValuesList(key, clazz, null);
+    }
+
+    /**
+     * Gets the list value of the given key, casting the list elements to {@code Class<T>} or returning the default list value if null.
+     * This is useful to avoid having casts in client code, though the effect is the same.
+     *
+     * @param key   the key
+     * @param clazz the non-null class to cast the list value to
+     * @param defaultValue what to return if the value is null
+     * @param <T>   the type of the class
+     * @return the list value of the given key, or the default list value if the instance does not contain this key.
+     * @throws ClassCastException if the value of the given key is not of type T
+     * @since 3.5
+     */
+    public <T> List<T> getList(final Object key, final Class<T> clazz, final List<T> defaultValue) {
+        notNull("defaultValue", defaultValue);
+        notNull("clazz", clazz);
+        return constructValuesList(key, clazz, defaultValue);
+    }
+
+    /**
```

Comment:
Generally we don't add javadoc-style comments for private methods, as they are not included in the public API docs anyway.  If you think a comment is necessary for code maintainers, just use `//` style.

Reply:
I have changed the comment to the `//` style.